fs/span-manager/span_manager.go

```diff
@@ -113,19 +116,23 @@ type spanInfo struct {
 	spanIndexInBuf []compression.Offset
 }
 
-func New(ztoc *ztoc.Ztoc, r *io.SectionReader, cache cache.BlobCache, cacheOpt ...cache.Option) *SpanManager {
+func New(ztoc *ztoc.Ztoc, r *io.SectionReader, cache cache.BlobCache, retries int, cacheOpt ...cache.Option) *SpanManager {
 	index, err := compression.NewGzipZinfo(ztoc.CompressionInfo.Checkpoints)
 	if err != nil {
 		return nil
 	}
 	spans := make([]*span, ztoc.CompressionInfo.MaxSpanID+1)
 	m := &SpanManager{
-		cache:    cache,
-		cacheOpt: cacheOpt,
-		index:    index,
-		r:        r,
-		spans:    spans,
-		ztoc:     ztoc,
+		cache:                    cache,
+		cacheOpt:                 cacheOpt,
+		index:                    index,
+		r:                        r,
+		spans:                    spans,
+		ztoc:                     ztoc,
+		maxSpanVerificationTries: retries + 1,
+	}
+	if m.maxSpanVerificationTries <= 0 {
+		m.maxSpanVerificationTries = defaultRetries
```

Comment:
I feel this is a little confusing to me: try sometimes = retry+1 and sometimes try = retry

```
try = retry + 1
if try <= 0 (retry <= -1)
then try = default retry
```
which means we're using `retry` differently: for 1st, retry is the extra tries; for 2nd, retry is the total tries.

I suggest we only use `retry`: if retry >= 0, keep the value; if retry <0, use the default.
when fetch, we use `i < retry + 1` (minimum 1 fetch, plus x retry)